File: frontend/angular/src/app/services/bookfiction.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Bookfiction} from "../domain/bookfiction";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class BookfictionService {
  constructor(private httpClient: HttpClient) {
  }

  createBookfiction(bookfiction: Bookfiction): Observable<Bookfiction> {
    return this.httpClient.post<Bookfiction>("/bookfiction/create", bookfiction);
  }

  getBookfictionGenres(): Observable<string[]> {
    return this.httpClient.get<string[]>("/bookfiction/genres");
  }

  updateBookfiction(bookfiction: Bookfiction): Observable<void> {
    return this.httpClient.put<void>('bookfiction/edit', bookfiction);
  }
}

File: backend/src/main/java/com/realdolmen/backend/service/BookfictionService.java
package com.realdolmen.backend.service;

import com.realdolmen.backend.domain.Bookfiction;

public interface BookfictionService extends CrudService<Bookfiction, Long> {

    Bookfiction create(Bookfiction bookfiction);

    Bookfiction findById(Long articleId);
}
